crates/swc_common/src/syntax_pos/analyze_source_file.rs

```diff
@@ -240,7 +80,7 @@ fn analyze_source_file_generic(
             let pos = BytePos::from_usize(i) + output_offset;
 
             match byte {
-                b'\n' => {
+                b'\n' | b'\r' => {
```

Comment:
I think we need to handle `\r\n` as a single line ending and not two line endings (unless that's been normalized as some prior point I'm not seeing).

Reply:
Did it, and I'll add some tests for it.